### ISSUE TITLE
source check does not recognize different shell

### ISSUE THREAD

**Issue by SimonHeimberg** [koalaman/shellcheck#1764]
#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related
- checked on 0.7.0 (stable and latest on docker)

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/dash
#  file d.sh
. b.sh

#!/bin/bash
# file b.sh
echo "${PATH//:/ }"

#command to check the files:
shellcheck d.sh b.sh
```

#### Here's what shellcheck currently says:
no error

#### Here's what I wanted or expected to see:
report that the sourced `b.sh` has a different shell type than the callling `d.sh`.

**Comment by SimonHeimberg** [comment]
Sourcing a bash file (b.sh) from a sh file (d.sh) will often result in problems (above: `bad substitution`).

The other way around (sourcing a sh file from a bash file) is no problem. (There may be some edge cases.)
